src/components/button/button.js

```diff
@@ -33,12 +34,52 @@ Button.prototype.handleKeyDown = function (event) {
   }
 }
 
+/**
+* If the click quickly succeeds a previous click then nothing will happen.
+* This stops people accidentally causing multiple form submissions by
+* double clicking buttons.
+*/
+Button.prototype.debounce = function (event) {
+  var target = event.target
+  // Check the button that is clicked on has the preventDoubleClick feature enabled
+  if (target.attributes && target.getAttribute('data-prevent-double-click') !== 'true') {
```

Comment:
If the target doesn't have any attributes (not sure how likely that is), this won't evaluate the second statement. I think it needs to be flipped:

```suggestion
  if (!target.attributes || target.getAttribute('data-prevent-double-click') !== 'true') {
```

Reply:
Actually, I think getAttribute is generally safe to use, and will just return `null` if the attribute isn't set, so not sure the first check is necessary? In which case this could just be

```suggestion
  if (target.getAttribute('data-prevent-double-click') !== 'true') {
```

Reply:
I've been guarding them since if you try to use 'getAttribute' on document it will throw.

Not sure this makes sense here though.